app/controllers/portfolio_controller.rb

```diff
@@ -0,0 +1,18 @@
+get '/portfolio' do
+  update_quotes
+  erb :portfolio
+end
+
+post '/buy' do
+  Transaction.create(buy_quantity: params[:quantity].to_i, buy_price: params[:price].to_f, buy_datetime: DateTime.parse(params[:datetime]), index_id: 2, portfolio_id: current_portfolio.id, stock_id: params[:stock_id])
+  redirect '/portfolio'
+end
+
+post '/sell' do
```

Comment:
it is not clear what is going on here. Does sell return an error message? You should probably have it raise an error.